[nats helm] use config checksum in addition to reloader

caleblloyd opened the pull request:
- Resolves #443 
- The Configuration file remains unchanged, but is moved out of `templates` and into `files` sub-directory so that it can be accessed by Helm `.Files` helper

wallyqs commented on helm/charts/nats/values.yaml:
```diff
@@ -76,9 +76,17 @@ nats:
 
+  # Toggle to enable sharing process namespace in the Pod
+  # Useful for passing signals to NATS server
+  shareProcessNamespace: false
```
maybe also add a note to the readme under misc that this is needed to re-enable? https://github.com/nats-io/k8s/blob/main/helm/charts/nats/README.md#configuration-reload-sidecar

caleblloyd replied:
The conditional to actually enable `shareProcessNamespace` on the pod does also consider the reloader - `{{- if or .Values.nats.shareProcessNamespace .Values.reloader.enabled }}`...   I can update the comment to mention that

wallyqs replied:
ah ok, I see now. I think it is fine as is then

wallyqs reviewed (approved):
LGTM

caleblloyd commented:
@wallyqs I added a section about the Config Checksum to the README

I also updated the Config Reloader documentation - and stated that it is Deprecated.  Do you agree that the Config Reloader should be deprecated and potentially removed in a further release?

wallyqs reviewed (approved):
LGTM
